### PR TITLE
add-global-status-cmd

### PR DESCRIPTION
**Pull request by WinnieXU1**
#1. add "infrasim global status" command
this cmd will list all the nodes information in the workspace, including all the process it has and all 
the ports it occupied. 
global_status1
global_status2
global_status3
2. add a test for this command

### CODE REVIEW

**Comment by chenge3** [comment]
@WinnieXU1 
1. Shall we add vDell server's "racadm" status into global_status output as below? 
[ 2976   ] full-racadm is running

**Comment by WinnieXU1** [comment]
@chenge3 have added "racadm" status~

File: infrasim/global_status.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import os
```
Comment (turtle-fly) [inline infrasim/global_status.py]: Hi, @WinnieXU1 ,

At first glance, I think global status is a behavior and doesn't need to be a module. 
To think further, you can actually make a module which is responsible for global monitoring:
- capture process status;
- capture port utilization;
- ...
I'm looking forward to a better design (could be a incomplete implementation) on this module.

About to display the information, vagrant shows a more concise style which worth learning.

**Comment by WinnieXU1** [comment]
global_status1
global_status2
global_status3

**Comment by WinnieXU1** [comment]
global_status3
global_status2
global_status1

**Comment by XiaowenJiang** [comment]
cool

File: test/functional/test_cli_commands.py [hunk 2]
```diff
@@ -395,8 +399,8 @@ def test_infrasim(self):
         """
         CLI test: "infrasim -h" navigates next level command usage
         """
-        p = r'\{([a-zA-Z,]+)\}'
-        cmd_next_level = ["node", "chassis", "config", "init", "version"]
+        p = r'\{([a-zA-Z_,]+)\}'
+        cmd_next_level = ["node", "chassis", "config", "init", "version", "global_status"]
```
Comment (fub2) [inline test/functional/test_cli_commands.py]: [minor]embedding a dash looks less consistent than other argument, like node, chassis, config, etc. can we make it a "infrasim global status"

**Comment by turtle-fly** [comment]
@fub2 expand your idea to:
- infrasim global [status | port | pid | namespace | network | ...]
for more runtime situation descriptions

@xiar @XiaowenJiang @chenge3 @JuneZhou19 @WinnieXU1 how do you think about this

**Comment by emcinfrasim** [comment]
```
******* Functional Test ******* 
TestName: test_node_global_status_ipmi_console 
ErrorType: exceptions.AssertionError 
ErrorMessage: 
-------------------- >> begin captured stdout << ---------------------
[ 25667  ] test1-socat starts to run
[ 25670  ] test1-bmc starts to run
[ 25679  ] test1-node is running
[ 25690  ] test1-racadm starts to run

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
root: INFO: Creating drive: 
root: INFO: /usr/bin/socat pty,link=/home/infrasim/.infrasim/test1/.pty0,waitslave unix-listen:/home/infrasim/.infrasim/test1/.serial,fork
root: INFO: ipmi_sim -c /home/infrasim/.infrasim/test1/etc/vbmc.conf -f /home/infrasim/.infrasim/test1/data/dell_r730.emu -n -s /home/infrasim/.infrasim/test1/data
root: INFO: racadmsim test1 0.0.0.0 10022 admin admin auto
--------------------- >> end captured logging << --------------------- 
TestName: test_nodes_global_status_start_stop_destroy 
ErrorType: exceptions.AssertionError 
ErrorMessage: 
-------------------- >> begin captured stdout << ---------------------
[ 25776  ] test1-socat starts to run
[ 25779  ] test1-bmc starts to run
[ 25788  ] test1-node is running
[ 25799  ] test1-racadm starts to run

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
root: INFO: Creating drive: 
root: INFO: /usr/bin/socat pty,link=/home/infrasim/.infrasim/test1/.pty0,waitslave unix-listen:/home/infrasim/.infrasim/test1/.serial,fork
root: INFO: ipmi_sim -c /home/infrasim/.infrasim/test1/etc/vbmc.conf -f /home/infrasim/.infrasim/test1/data/dell_r730.emu -n -s /home/infrasim/.infrasim/test1/data
root: INFO: racadmsim test1 0.0.0.0 10022 admin admin auto
--------------------- >> end captured logging << --------------------- 
```

**Comment by emcinfrasim** [comment]
```
******* Functional Test ******* 
TestName: test_node_global_status_ipmi_console 
ErrorType: exceptions.AssertionError 
ErrorMessage: 
-------------------- >> begin captured stdout << ---------------------
[ 8989   ] test1-socat starts to run
[ 8992   ] test1-bmc starts to run
[ 9001   ] test1-node is running
[ 9012   ] test1-racadm starts to run

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
root: INFO: Creating drive: 
root: INFO: /usr/bin/socat pty,link=/home/infrasim/.infrasim/test1/.pty0,waitslave unix-listen:/home/infrasim/.infrasim/test1/.serial,fork
root: INFO: ipmi_sim -c /home/infrasim/.infrasim/test1/etc/vbmc.conf -f /home/infrasim/.infrasim/test1/data/dell_r730.emu -n -s /home/infrasim/.infrasim/test1/data
root: INFO: racadmsim test1 0.0.0.0 10022 admin admin auto
--------------------- >> end captured logging << --------------------- 
TestName: test_nodes_global_status_start_stop_destroy 
ErrorType: exceptions.AssertionError 
ErrorMessage: 
-------------------- >> begin captured stdout << ---------------------
[ 9108   ] test1-socat starts to run
[ 9111   ] test1-bmc starts to run
[ 9120   ] test1-node is running
[ 9131   ] test1-racadm starts to run

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
root: INFO: Creating drive: 
root: INFO: /usr/bin/socat pty,link=/home/infrasim/.infrasim/test1/.pty0,waitslave unix-listen:/home/infrasim/.infrasim/test1/.serial,fork
root: INFO: ipmi_sim -c /home/infrasim/.infrasim/test1/etc/vbmc.conf -f /home/infrasim/.infrasim/test1/data/dell_r730.emu -n -s /home/infrasim/.infrasim/test1/data
root: INFO: racadmsim test1 0.0.0.0 10022 admin admin auto
--------------------- >> end captured logging << --------------------- 
```

File: test/functional/test_cli_commands.py [hunk 3]
```diff
@@ -395,8 +399,8 @@ def test_infrasim(self):
         """
         CLI test: "infrasim -h" navigates next level command usage
         """
-        p = r'\{([a-zA-Z,]+)\}'
-        cmd_next_level = ["node", "chassis", "config", "init", "version"]
+        p = r'\{([a-zA-Z_,]+)\}'
```
Comment (turtle-fly) [inline test/functional/test_cli_commands.py]: won't need the underscore now

**Comment by turtle-fly** [comment]
Retest this please

File: test/functional/test_cli_commands.py (hunk 3 above)
Comment (WinnieXU1) [inline test/functional/test_cli_commands.py]: have removed that

Comment (WinnieXU1) [inline test/functional/test_cli_commands.py]: have removed that

**Pull request merged**